fix(ios): fix compatability with iOS 9

ewanharris opened the pull request:
Fixes TIMOB-27292

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-27292

This adds back iOS 9 into our supported version range. I haven't touched the xcode version as I'm not fully clear on what it means to add back xcode support, I'm guessing we need to revert the `IS_XCODE_9` removal?

ewanharris committed:
fix(ios): fix compatability with iOS 9

Fixes TIMOB-27292

build commented:
<!--
  1 failure:  Tests have failed...
  0 warning: 
  3 messages
  1 markdown notices
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-11091/1/artifact/dist/mobilesdk-8.2.0.v20190729082431-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:x: 3 tests have failed
There are 3 tests failing and 470 skipped out of 3728 total tests.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  </tbody>
</table>

### Tests:

<table><tr><th>Classname</th><th>Name</th><th>Time</th><th>Error</th></tr>
<tr><td>android.emulator.Titanium.UI.Layout</td><td>TIMOB-23305</td><td>10.214</td><td><pre>Error: timeout of 5000ms exceeded
at Titanium.&lt;anonymous&gt; (/ti-mocha.js:4326:18)</pre></td></tr>
<tr><td>android.emulator.Titanium.UI.Layout</td><td>fourPins</td><td>6.446</td><td><pre>Error: timeout of 5000ms exceeded
at Titanium.&lt;anonymous&gt; (/ti-mocha.js:4326:18)</pre></td></tr>
<tr><td>android.emulator.Titanium.UI.Layout</td><td>undefinedLeft</td><td>5.012</td><td><pre>Error: timeout of 5000ms exceeded
at Titanium.&lt;anonymous&gt; (/ti-mocha.js:4326:18)</pre></td></tr>
</table>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8908d61d06fe96e10f8b0212b6265c9cb6cb5bf5
</p>

vijaysingh-axway commented on iphone/package.json:
```diff
@@ -20,7 +20,7 @@
 	"minWatchosVersion": "2.0",
 	"vendorDependencies": {
 		"xcode": ">=9.0 <=11.x",
-		"ios sdk": ">=10.0 <=13.x"
```
@cb1kenobi  In my [PR-10932](https://github.com/appcelerator/titanium_mobile/pull/10932), I have made changes on basis of our previous history to support minimum iOS SDK and minimum XCode version  from [here](https://github.com/appcelerator/titanium_mobile/pull/10105) (to support iOS 12). So I made minimum Xcode to Xcode 9.0 and minimum iOS SDK to 10.0. Did I missed any change in [PR-10932](https://github.com/appcelerator/titanium_mobile/pull/10932)?. I think for pulling minimum iOS version [here](https://github.com/appcelerator/titanium_mobile/blob/a48d707984e88e92f05287dcd3c2588025c9c9f2/build/lib/packager.js#L88)(), we should use [minIosVersion](https://github.com/appcelerator/titanium_mobile/blob/98fa81bb2d947b4c3010a73645e5f414da7d9605/iphone/package.json#L19)(if it requires minimum iOS target as @ewanharris  as mentioned in ticket).

cb1kenobi replied:
@vijaysingh-axway please see my comment in the [ticket](https://jira.appcelerator.org/browse/TIMOB-27292?focusedCommentId=450137&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-450137).

ewanharris committed:
build(ios): correct property used for minimum ios version

ewanharris commented:
@cb1kenobi, @vijaysingh-axway fixed the build scripts to reference `minIosVersion`. Shall I also go ahead and work on adding back the xcode 8 compat?

vijaysingh-axway commented on build/lib/packager.js:
```diff
@@ -85,7 +83,7 @@ function determineBabelOptions() {
 	// Now pull out min IOS target
 	// eslint-disable-next-line security/detect-non-literal-require
-	const minSupportedIosSdk = version.parseMin(require(path.join(ROOT_DIR, 'iphone/package.json')).vendorDependencies['ios sdk']);
+	const minSupportedIosSdk = require(path.join(ROOT_DIR, 'iphone/package.json')).minIosVersion;
```
@ewanharris Can you please change const  name 'minSupportedIosSdk' to 'minSupportedIosVersion' because it is a bit confusing.

vijaysingh-axway commented on build/scons-xcode-project-build.js:
```diff
@@ -53,7 +51,7 @@ async function generateIndexJSON(dirToTraverse) {
 function determineBabelOptions() {
 	// eslint-disable-next-line security/detect-non-literal-require
-	const minSupportedIosSdk = version.parseMin(require(path.join(ROOT_DIR, 'iphone/package.json')).vendorDependencies['ios sdk']);
+	const minSupportedIosSdk = require(path.join(ROOT_DIR, 'iphone/package.json')).minIosVersion;
```
Same as above. Please change name.

vijaysingh-axway commented:
> @cb1kenobi, @vijaysingh-axway fixed the build scripts to reference `minIosVersion`. Shall I also go ahead and work on adding back the xcode 8 compat?

@ewanharris Please change variable name as mentioned in comment. Please go ahead on adding back the xcode 8. I'll create a new ticket to  make minimum Xcode change and minimum iOS SDK change in SDK 9.

ewanharris committed:
refactor: pull minIosVersion off package.json object

ewanharris committed:
fix(ios): add back ios 11 constants to code

Rename from IS_XCODE_9 to IS_IOS_SDK_11 per the discussion in https://github.com/appcelerator/titanium_mobile/pull/10932

ewanharris commented:
@vijaysingh-axway, updated to both those. Also, per yours and Jan's discussion [here](https://github.com/appcelerator/titanium_mobile/pull/10932/files/a6ac8647fc639a121a9c390fdc3f6f0acd175c52#r304722755) I changed the constant from `IS_XCODE_9` to `IS_SDK_IOS_11`

vijaysingh-axway commented on iphone/package.json:
```diff
@@ -20,7 +20,7 @@
 	"minWatchosVersion": "2.0",
 	"vendorDependencies": {
 		"xcode": ">=9.0 <=11.x",
```
@ewanharris  Can you revert minimum Xcode version to 8.0 as well.

ewanharris replied:
@vijaysingh-axway, dang I missed that 😅 Just pushed it now

vijaysingh-axway reviewed (changes requested):
@ewanharris One minor change required. Everything else looks good.

ewanharris committed:
fix(ios): correct minimum xcode version

vijaysingh-axway reviewed (approved):
CR passed.

vijaysingh-axway committed:
Merge branch 'master' into TIMOB-27292

vijaysingh-axway committed:
Merge branch 'master' into TIMOB-27292

ssekhri commented:
FR Passed.
App builds and launches successfully on iOS 9 simulator
Verified on:
Mac OS 10.14.6
Ti SDK: 8.2.0.v20190807170125
Appc CLI: 7.1.1-master.2
Node: 10.5.0
JDK: 10.0.2
Xcode: 11.0 beta 5
Studio: 5.1.3.201907112159

Commit:
Merge branch 'master' into TIMOB-27292

Commit:
Merge branch 'master' into TIMOB-27292

Commit:
Merge branch 'master' into TIMOB-27292
